spec/bolt/transport/local_spec.rb

```diff
@@ -29,10 +34,169 @@
+          expect(runner.run_command(target, "stat -c %U #{dest}", '_run_as' => user)['stdout'].chomp).to eq(user)
+          expect(runner.run_command(target, "stat -c %G #{dest}", '_run_as' => user)['stdout'].chomp).to eq(user)
+        end
+
+        runner.run_command(target, "rm #{dest}", sudoable: true, run_as: user)
+      end
+    end
+
+    context "as user with no password" do
+      let(:config) {
+        mk_config('run-as' => 'root')
+      }
+      let(:target) { make_target }
+
+      it "returns a failed result when a temporary directory is created" do
+        contents = "#!/bin/sh\nwhoami"
+        with_tempfile_containing('script test', contents) do |file|
+          expect {
+            runner.run_script(target, file.path, [])
+          }.to raise_error(Bolt::Node::EscalateError,
+                           "Sudo password for user #{user} was not provided for #{host_and_port}")
+        end
+      end
+    end
+  end
+
+  context 'with large input and output' do
+    let(:file_size) { 100 }
+    let(:str) { (0...1024).map { rand(65..90).chr }.join }
+    let(:arguments) { { 'input' => str * file_size } }
+    let(:ruby_task) do
+      <<~TASK
+        require 'json'
+
+        input = JSON.parse(STDIN.read)['input']
+        STDERR.print input
+        STDOUT.print input
+        TASK
+    end
+
+    it "runs with large input and captures large output" do
+      with_task_containing('big_kahuna', ruby_task, 'stdin', '.rb') do |task|
+        result = runner.run_task(target, task, arguments).value
+        puts result
+        expect(result['_output'].bytesize).to eq(file_size * 1024)
+        # Ensure the strings are the same
+        expect(result['_output'][-1024..-1]).to eq(str)
+      end
+    end
+
+    context 'with run-as', sudo: true do
+      let(:config) {
+        mk_config('sudo-password' => password, 'run-as' => 'root')
+      }
+      let(:target) { make_target }
+
+      it "runs with large input and output" do
+        with_task_containing('big_kahuna', ruby_task, 'stdin', '.rb') do |task|
+          result = runner.run_task(target, task, arguments).value
+          expect(result['_output'].bytesize).to eq(file_size * 1024)
+          expect(result['_output'][-1024..-1]).to eq(str)
+        end
+      end
+    end
+
+    context 'with slow input' do
+      let(:ruby_task) do
+        <<~TASK
+          while true
+            begin
+              input = STDIN.readpartial(1024)
+              sleep(0.2)
+              STDERR.print input
+              STDOUT.print input
+            rescue EOFError
+              break
+            end
+          end
+          TASK
+      end
+
+      it "runs with large input and captures large output" do
+        with_task_containing('slow_and_steady', ruby_task, 'stdin', '.rb') do |task|
+          result = runner.run_task(target, task, arguments).value
+          expect(result['input'].bytesize).to eq(file_size * 1024)
```

Comment:
input is nil which causes an error here. Looking at the code that is what I would expect.

Reply:
I was seeing `input` because we pass params in as JSON but don't parse them in the task, so the return is also json which bolt makes the result object. I'm not sure why you wouldn't expect `input`?

Reply:
ah that makes sense. But that won't differentiate from the edge case where stdin gets printed by sudo when the task exits without reading it.

It also looks like this task will fail to print the last chunk if the eof isn't exactly on a boundary.

Reply:
Updated to 1011kb, so it's not a multiple of 4kb that get read in the loop

Reply:
That sounds like the edge case where when a task doesn't consume stdin and is wrapped in sudo it gets printed to stdout by sudo.